kimchi/src/circuits/gate.rs

```diff
@@ -273,6 +298,24 @@ impl<F: FftField> LookupInfo<F> {
     }
 }
 
+/// Specifies the relative position of gates and the fixed lookup table (if applicable) that a
+/// given lookup configuration should apply to.
+pub struct GatesLookupSpec {
+    /// The set of positions relative to an active gate where a lookup configuration applies.
+    pub gate_positions: HashSet<(GateType, CurrOrNext)>,
+    /// The fixed lookup table that should be used for these lookups, if applicable.
+    pub gate_lookup_table: Option<GateLookupTable>,
+}
+
+/// Specifies mapping from positions defined relative to gates into lookup data.
+pub struct GatesLookupMaps {
+    /// Enumerates the selector that should be active for a particular gate-relative position.
+    pub gate_selector_map: HashMap<(GateType, CurrOrNext), usize>,
+    /// Enumerates the fixed tables that should be used for lookups in a particular gate-relative
+    /// position.
```

Comment:
No need to break that comment for a single word

Reply:
As above, the line is already too long IMO

Reply:
that sounds like ideology :P